### PR TITLE
Bluetooth Mesh badge demo for the reel board

### PR DESCRIPTION
**Pull request by jhedberg**
I'm uploading this as a PR so that those with reel boards can try it out.

The app in question starts off as a regular Bluetooth GATT peripheral application. I recommend installing the "nRF Connect" app on your phone (available both for Android and iOS) to access the service that the app exposes.

1. Scan for devices and look for "reel board"
2. Connect to the device. You'll see a single service - select it
3. Request to write to the characteristic by pressing on the upward pointing arrow symbol
4. Select "Text" to enter text instead of hex
5. Enter your name (or any other random data). Multiple words separated by spaces are possible. The font used enables three rows of up to 12 characters wide text. You can force line breaks with a comma.
6. Press "Send" - this will trigger pairing since this is a protected characteristic. The passkey for the pairing will be shown on the board's display.
7. Once pairing is complete the board will show the new name. If you're not happy with it you can try writing something else.
8. When you're happy with the text, disconnect from the board (exit the app or go back to the device scan page)
9. Once disconnected the board switches over to Bluetooth Mesh mode, and you can't connect to it anymore over GATT. If you configure multiple boards like this they can communicate with each other over mesh by pressing their user button.

To reset a board to its initial state (disable mesh, erase the name and make it connectable over GATT):

1. Keep the user button pressed when powering on (or press the reset button when powered)
2. Wait until "Reseting Device" is shown

### CODE REVIEW

**Comment by jfischer-phytec-iot** [comment]
@jhedberg Thank you, nice example. I will try to replace the font with Clear Sans.

**Review by nashif — CHANGES_REQUESTED** [review]
Needs documentation

**Comment by jhedberg** [comment]
> Needs documentation

@nashif done

File: samples/boards/reel_board/mesh_badge/README.rst [hunk 1]
```diff
@@ -0,0 +1,43 @@
+.. _mesh_badge:
+
+Mesh Badge
+##########
+
+Overview
+********
+
+A sample app for the reel board that showcases Bluetooth Mesh
```
Comment (dbkinder) [inline samples/boards/reel_board/mesh_badge/README.rst]:
```suggestion
This sample app for the reel board showcases Bluetooth Mesh.
```

File: samples/boards/reel_board/mesh_badge/README.rst [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+The app starts off as a regular Bluetooth GATT peripheral application. It's
```
Comment (dbkinder) [inline samples/boards/reel_board/mesh_badge/README.rst]: Delete ``It's recommended to" and start the sentence with:

    Install the ...

File: samples/boards/reel_board/mesh_badge/README.rst [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+recommended to install the "nRF Connect" app on your phone (available both for
+Android and iOS) to access the service that the app exposes.
+
+Steps to set up
+***************
+
+#. Scan for devices and look for "reel board"
```
Comment (dbkinder) [inline samples/boards/reel_board/mesh_badge/README.rst]:
```suggestion
#. On your phone, use the nRF Connect app to scan for devices and look for "reel board"
```

File: samples/boards/reel_board/mesh_badge/README.rst [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+#. Connect to the device. You'll see a single service - select it
+#. Request to write to the characteristic by pressing on the upward pointing
+   arrow symbol
+#. Select "Text" to enter text instead of hex
+#. Enter your name (or any other random data). Multiple words separated by
+   spaces are possible. The font used enables three rows of up to 12 characters
```
Comment (dbkinder) [inline samples/boards/reel_board/mesh_badge/README.rst]:
```suggestion
   spaces are possible. The font used on the reel display allows three rows of up to 12 characters
```

File: samples/boards/reel_board/mesh_badge/README.rst [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+   wide text. You can force line breaks with a comma.
+#. Press "Send" - this will trigger pairing since this is a protected
+   characteristic. The passkey for the pairing will be shown on the board's
+   display.
```
Comment (dbkinder) [inline samples/boards/reel_board/mesh_badge/README.rst]: I presume you need to do something in the app to complete the paring using the passkey?

File: samples/boards/reel_board/mesh_badge/README.rst [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+#. Once pairing is complete the board will show the new name. If you're not
```
Comment (dbkinder) [inline samples/boards/reel_board/mesh_badge/README.rst]:
```suggestion
#. Once pairing is complete the board will show the text you sent. If you're not
```

File: samples/boards/reel_board/mesh_badge/README.rst [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+   happy with it you can try writing something else.
+#. When you're happy with the text, disconnect from the board (exit the app or
+   go back to the device scan page)
+#. Once disconnected the board switches over to Bluetooth Mesh mode, and you
+   can't connect to it anymore over GATT. If you configure multiple boards like
+   this they can communicate with each other over mesh by pressing their user
+   button.
```
Comment (dbkinder) [inline samples/boards/reel_board/mesh_badge/README.rst]: ... and what will happen after they "communicate with each other"?

File: samples/boards/reel_board/mesh_badge/README.rst [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+To reset a board to its initial state (disable mesh, erase the name and make it
```
Comment (dbkinder) [inline samples/boards/reel_board/mesh_badge/README.rst]:
```suggestion
To reset a board to its initial state (disable mesh, erase the stored text, and make it
```

File: samples/boards/reel_board/mesh_badge/README.rst [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+##########
+
+Overview
+********
+
+A sample app for the reel board that showcases Bluetooth Mesh
+
+The app starts off as a regular Bluetooth GATT peripheral application. It's
+recommended to install the "nRF Connect" app on your phone (available both for
+Android and iOS) to access the service that the app exposes.
```
Comment (dbkinder) [inline samples/boards/reel_board/mesh_badge/README.rst]: Can you do this board setup with a program running on your laptop (or only with your phone)?

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: samples/boards/reel_board/mesh_badge/README.rst (hunk 1 above)
Comment (jhedberg) [reply samples/boards/reel_board/mesh_badge/README.rst]: done

File: samples/boards/reel_board/mesh_badge/README.rst (hunk 2 above)
Comment (jhedberg) [reply samples/boards/reel_board/mesh_badge/README.rst]: done

File: samples/boards/reel_board/mesh_badge/README.rst (hunk 3 above)
Comment (jhedberg) [reply samples/boards/reel_board/mesh_badge/README.rst]: done

File: samples/boards/reel_board/mesh_badge/README.rst (hunk 4 above)
Comment (jhedberg) [reply samples/boards/reel_board/mesh_badge/README.rst]: done

File: samples/boards/reel_board/mesh_badge/README.rst (hunk 5 above)
Comment (jhedberg) [reply samples/boards/reel_board/mesh_badge/README.rst]: yes, you need to enter it in the phone - I added this text

File: samples/boards/reel_board/mesh_badge/README.rst (hunk 6 above)
Comment (jhedberg) [reply samples/boards/reel_board/mesh_badge/README.rst]: done

File: samples/boards/reel_board/mesh_badge/README.rst (hunk 7 above)
Comment (jhedberg) [reply samples/boards/reel_board/mesh_badge/README.rst]: I added text to describe this

File: samples/boards/reel_board/mesh_badge/README.rst (hunk 8 above)
Comment (jhedberg) [reply samples/boards/reel_board/mesh_badge/README.rst]: done

File: samples/boards/reel_board/mesh_badge/README.rst (hunk 9 above)
Comment (jhedberg) [reply samples/boards/reel_board/mesh_badge/README.rst]: yes, I added a note about that, but I'm not planning to provide detailed instructions for every possible OS, Bluetooth stack and GATT client app :)

**Review by jfischer-no — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**